db/migrate/20210831154710_de_duplicate_response_sets.rb

```diff
@@ -0,0 +1,52 @@
+class DeDuplicateResponseSets < ActiveRecord::Migration[6.1]
+
+  # NOTE: up/down is written such that rollback can be run immediately to undo the up changes and vica versa,
+  # preserving which response set is the 'non-dup.'
+  # If up runs, then down runs, and then response sets are updated some other way (e.g. via the UI),
+  # up may not set the same response set as the non-dup as the first time.
+  def up
+    Loan.find_each do |loan|
+      kinds_present = loan.response_sets.map { |r| r.kind }.uniq
+      kinds_present.each do |kind|
+        dup_responses = loan.response_sets.select { |r| r.kind == kind }
+        sorted_dups = dup_responses.sort_by { |r| [r.updated_at, r.id] }
+        last_index_to_change = sorted_dups.count - 2 # last index is num_dups-1. Don't change last index.
+        for i in (0..last_index_to_change) do
+          dup = sorted_dups[i]
+          new_kind = "#{dup.kind}-dup-#{i}"
+          log_str = "For loan #{loan.id}: Updating response set id #{dup.id} (last updated at #{dup.updated_at}) kind from #{dup.kind} to #{new_kind}"
+          say_with_time log_str do
+            dup.update(kind: new_kind)
```

Comment:
Oh, this should be `update!` or you could end up with silent failures.

Reply:
thanks!